lib/index.js

```diff
@@ -3,28 +3,43 @@
 const Fs = require('fs');
 const Path = require('path');
 const Util = require('util');
+const Glob = require('glob');
 
 const internals = {
     readDir: Util.promisify(Fs.readdir),
-    testDefault: /\.(js)$/
+    testDefault: /\.(js)$/,
+    glob: Glob.sync
 };
 
 module.exports = {
     pkg: require('../package.json'),
     register: async (server, options) => {
 
-        const fileList = await internals.readDir(options.dir);
         const fileTest = options.test || internals.testDefault;
 
-        fileList.forEach((file) => {
-
-            if (file.match(fileTest)) {
-                const modulePath = Path.join(options.dir, file);
-                const routeFile = require(modulePath);
-
-                routeFile.routes(server);
-            }
-        });
+        if ('glob' in options && options.glob === true) {
```

Comment:
Is the `'glob' in options` check necessary? Can the next check be sufficient enough?

Reply:
This does indeed look like checking the same thing twice